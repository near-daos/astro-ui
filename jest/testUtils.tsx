import {
  RenderResult,
  render as reactTestingLibRender,
} from '@testing-library/react';
import { ReactNode } from 'react';
import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context';

function createMockRouter(router?: Partial<NextRouter>): NextRouter {
  return {
    basePath: '',
    pathname: '/',
    route: '/',
    query: {},
    asPath: '/',
    back: jest.fn(),
    beforePopState: jest.fn(),
    prefetch: jest.fn(),
    push: jest.fn(),
    reload: jest.fn(),
    replace: jest.fn(),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
    isLocaleDomain: false,
    isReady: true,
    defaultLocale: 'en',
    domainLocales: [],
    isPreview: false,
    ...router,
  };
}

export function render(
  component: ReactNode,
  router?: Partial<NextRouter>
): RenderResult {
  return reactTestingLibRender(
    <RouterContext.Provider value={createMockRouter(router)}>
      {component}
    </RouterContext.Provider>
  );
}
