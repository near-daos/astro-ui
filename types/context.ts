import { ProposalFeedItem, ProposalType } from 'types/proposal';
import { DAO } from 'types/dao';

export type UserPermissions = {
  isCanCreateProposals: boolean;
  isCanCreatePolicyProposals: boolean;
  allowedProposalsToCreate: ProposalPermissions;
  allowedProposalsToVote: ProposalPermissions;
};

export type ProposalPermissions = {
  [ProposalType.ChangeConfig]: boolean;
  [ProposalType.ChangePolicy]: boolean;
  [ProposalType.AddBounty]: boolean;
  [ProposalType.BountyDone]: boolean;
  [ProposalType.Transfer]: boolean;
  [ProposalType.Vote]: boolean;
  [ProposalType.RemoveMemberFromRole]: boolean;
  [ProposalType.AddMemberToRole]: boolean;
  [ProposalType.AddMemberToRole]: boolean;
  [ProposalType.SetStakingContract]: boolean;
  [ProposalType.UpgradeSelf]: boolean;
  [ProposalType.UpgradeRemote]: boolean;
  [ProposalType.FunctionCall]: boolean;
};

export type DaoContext = {
  userPermissions: UserPermissions;
  dao: DAO;
  policyAffectsProposals: ProposalFeedItem[];
};
