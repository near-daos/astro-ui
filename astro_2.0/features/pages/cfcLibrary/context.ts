import { createContext, useContext } from 'react';
import { DaoFeedItem } from 'types/dao';

interface Props {
  accountDaos: DaoFeedItem[];
  accountId: string;
  onUpdate: (templateId: string, daosCount: number) => void;
}

export const CfcLibraryContext = createContext<Props>({
  accountDaos: [],
  accountId: '',
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  onUpdate: () => {},
});

export const useCfcValues = (): Props => useContext<Props>(CfcLibraryContext);
