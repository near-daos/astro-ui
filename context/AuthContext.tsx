import { useRouter } from 'next/router';
import { createContext, FC, useContext, useState } from 'react';
import { useCookie } from 'react-use';

import { ALL_FEED_URL } from 'constants/routing';
import { ACCOUNT_COOKIE, DAO_COOKIE } from 'constants/cookies';

import { SputnikWalletError } from 'errors/SputnikWalletError';

import { NOTIFICATION_TYPES, showNotification } from 'features/notifications';
import { CookieService } from 'services/CookieService';
import { configService } from 'services/ConfigService';
import { WalletType } from 'types/config';
import { SputnikNearService } from 'services/sputnik';
import { SputnikWalletService } from 'services/sputnik/SputnikNearService/services/SputnikWalletService';

interface AuthContextInterface {
  accountId: string;
  login: () => Promise<void>;
  logout: () => Promise<void>;
  switchWallet: (walletType: WalletType) => Promise<void>;
}

const AuthContext = createContext<AuthContextInterface>({
  accountId: '',
  login: () => Promise.resolve(),
  logout: () => Promise.resolve(),
  switchWallet: () => Promise.resolve(),
});

export const AuthWrapper: FC = ({ children }) => {
  const router = useRouter();
  const [, , deleteAccountCookie] = useCookie(ACCOUNT_COOKIE);
  const [, , deleteDaoCookie] = useCookie(DAO_COOKIE);
  const [accountId, setAccountId] = useState<string>(
    CookieService.get(ACCOUNT_COOKIE)
  );
  const { nearConfig } = configService.get();

  async function login() {
    try {
      if (!window.nearService) {
        const walletService = new SputnikWalletService(nearConfig);

        window.nearService = new SputnikNearService(walletService);
      }

      await window.nearService?.signIn(nearConfig.contractName);

      const id = window.nearService?.getAccountId();

      if (id) {
        setAccountId(id);
      }
    } catch (err) {
      console.warn(err);

      if (err instanceof SputnikWalletError) {
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          description: err.message,
          lifetime: 20000,
        });
      }
    }
  }

  async function switchWallet(walletType: WalletType) {
    await window.nearService.switchWallet(walletType);
  }

  async function logout() {
    await window.nearService?.logout();

    setAccountId('');
    deleteAccountCookie();
    deleteDaoCookie();

    router.push(ALL_FEED_URL);
  }

  const data = {
    accountId,
    login,
    logout,
    switchWallet,
  };

  return <AuthContext.Provider value={data}>{children}</AuthContext.Provider>;
};

export function useAuthContext(): AuthContextInterface {
  return useContext(AuthContext);
}
