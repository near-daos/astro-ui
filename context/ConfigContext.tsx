import { createContext, FC, useContext, useEffect, useState } from 'react';
import axios from 'axios';
import { Config } from 'types/config';

interface ConfigContextInterface {
  config: Config | null;
}

const ConfigContext = createContext<ConfigContextInterface>({
  config: null,
});

export const ConfigContextProvider: FC = ({ children }) => {
  const [config, setConfig] = useState<Config | null>(null);

  useEffect(() => {
    const { CancelToken } = axios;
    const source = CancelToken.source();

    axios
      .get('/api/config', { cancelToken: source.token })
      .then(({ data }) => setConfig((data as unknown) as Config));

    return () => {
      source.cancel('Cancelled on unmount');
    };
  }, []);

  return (
    <ConfigContext.Provider value={{ config }}>
      {children}
    </ConfigContext.Provider>
  );
};

export function useConfigContext(): ConfigContextInterface {
  const context = useContext(ConfigContext);

  return context;
}
