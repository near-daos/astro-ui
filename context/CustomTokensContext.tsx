import reduce from 'lodash/reduce';
import { createContext, FC, useCallback, useContext, useState } from 'react';

import { SputnikHttpService } from 'services/sputnik';
import { Token } from 'types/token';

export type Tokens = Record<string, Token>;

interface CustomTokensContextInterface {
  tokens: Tokens;
  fetchAndSetTokens: () => Promise<void>;
  setTokens: (tokens: Token[]) => void;
}

/* eslint-disable @typescript-eslint/no-empty-function */
const CustomTokensContext = createContext<CustomTokensContextInterface>({
  tokens: {},
  fetchAndSetTokens: async () => {},
  setTokens: () => {},
});
/* eslint-enable @typescript-eslint/no-empty-function */

export const CustomTokensProvider: FC = ({ children }) => {
  const [tokens, setInnerTokens] = useState<Tokens>({});

  const setTokens = useCallback((tkns: Token[]) => {
    const resToken = reduce(
      tkns,
      (acc, token) => {
        const { tokenId, symbol } = token;

        acc[tokenId || symbol] = token;

        return acc;
      },
      {} as Tokens
    );

    setInnerTokens(resToken);
  }, []);

  async function fetchAndSetTokens() {
    const t = await SputnikHttpService.getTokens({
      limit: 1000,
    });

    setTokens(t);
  }

  const data = {
    tokens,
    setTokens,
    fetchAndSetTokens,
  };

  return (
    <CustomTokensContext.Provider value={data}>
      {children}
    </CustomTokensContext.Provider>
  );
};

export function useCustomTokensContext(): CustomTokensContextInterface {
  return useContext(CustomTokensContext);
}
