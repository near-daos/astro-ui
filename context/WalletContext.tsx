import { WalletType } from 'types/config';
import {
  WalletMeta,
  WalletService,
} from 'services/sputnik/SputnikNearService/walletServices/types';
import { createContext, FC, useCallback, useContext, useMemo } from 'react';
import { useBoolean } from 'react-use';

import { configService } from 'services/ConfigService';
import { CookieService } from 'services/CookieService';
import { ACCOUNT_COOKIE } from 'constants/cookies';
import { SputnikNearService } from 'services/sputnik';
import { useRouter } from 'next/router';
import { ConnectingWalletModal } from 'astro_2.0/features/Auth/components/ConnectingWalletModal';
import {
  PkAndSignature,
  useAvailableAccounts,
  useWallet,
  usePkAndSignature,
} from 'context/WalletContextHooks';
import { GA_EVENTS, sendGAEvent } from 'utils/ga';

export interface WalletContext {
  availableWallets: WalletMeta[];
  currentWallet: WalletType | null;
  accountId: string;
  connectingToWallet: boolean;
  login: (walletType: WalletType) => Promise<void>;
  logout: () => Promise<void>;
  switchAccount: (walletType: WalletType, accountId: string) => void;
  switchWallet: (walletType: WalletType) => void;
  availableAccounts: string[];
  pkAndSignature: PkAndSignature | null;
  // todo get rid of
  nearService: SputnikNearService | null;
}

const WalletContext = createContext<WalletContext>({} as WalletContext);

export const WrappedWalletContext: FC = ({ children }) => {
  const [
    currentWallet,
    availableWallets,
    getWallet,
    setWallet,
    removePersistedWallet,
  ] = useWallet();
  const availableAccounts = useAvailableAccounts(currentWallet);
  const pkAndSignature = usePkAndSignature(currentWallet);
  const { nearConfig } = configService.get();

  const [connectingToWallet, toggleConnection] = useBoolean(false);

  const router = useRouter();

  const signIn = useCallback(
    async (wallet: WalletService, contractName: string) => {
      toggleConnection(true);

      if (!wallet.isSignedIn()) {
        await wallet.signIn(contractName);
      }

      setWallet(wallet);

      toggleConnection(false);

      sendGAEvent({
        name: GA_EVENTS.SIGN_IN,
        accountId: wallet.getAccountId(),
      });
    },
    [setWallet, toggleConnection]
  );

  const login = useCallback(
    async (walletType: WalletType) => {
      const wallet = getWallet(walletType);

      if (!wallet) {
        return;
      }

      signIn(wallet, nearConfig.contractName);
    },
    [getWallet, nearConfig.contractName, signIn]
  );

  const logout = useCallback(async () => {
    sendGAEvent({
      name: GA_EVENTS.SIGN_OUT,
      accountId: currentWallet?.getAccountId() ?? '',
    });

    CookieService.remove(ACCOUNT_COOKIE);
    removePersistedWallet();
    availableWallets.forEach(wallet => wallet.logout());
    router.reload();
  }, [availableWallets, currentWallet, removePersistedWallet, router]);

  const switchAccount = useCallback(
    async (walletType: WalletType, accountId: string) => {
      // Currently its NEAR wallet specific only
      if (walletType !== WalletType.NEAR) {
        return;
      }

      const nearWallet = getWallet(WalletType.NEAR);

      if (!nearWallet) {
        return;
      }

      // eslint-disable-next-line no-underscore-dangle
      const keypair = await nearWallet
        .getKeyStore()
        .getKey(nearConfig.networkId, accountId);

      if (!keypair) {
        return;
      }

      const authData = {
        accountId,
        allKeys: [keypair.getPublicKey().toString()],
      };

      // new wallet instance will take the new auth_key and will reinit the account
      window.localStorage.setItem(
        'sputnik_wallet_auth_key',
        JSON.stringify(authData)
      );

      CookieService.set(ACCOUNT_COOKIE, accountId, {
        path: '/',
      });

      sendGAEvent({
        name: GA_EVENTS.SWITCH_ACCOUNT,
        accountId,
      });

      router.reload();
    },
    [getWallet, nearConfig.networkId, router]
  );

  const switchWallet = useCallback(
    async (walletType: WalletType) => {
      const selectedWallet = getWallet(walletType);

      if (!selectedWallet || !currentWallet) {
        return;
      }

      await signIn(selectedWallet, nearConfig.contractName);

      const currentWalletAccountId = currentWallet.getAccountId();
      const selectedWalletAccountId = selectedWallet.getAccountId();

      sendGAEvent({
        name: GA_EVENTS.SWITCH_WALLET,
        accountId: selectedWalletAccountId,
        params: {
          wallet: walletType,
          previousAccountId: currentWalletAccountId,
        },
      });

      if (currentWalletAccountId !== selectedWalletAccountId) {
        router.reload();
      }
    },
    [currentWallet, getWallet, nearConfig.contractName, router, signIn]
  );

  const walletContext = useMemo(() => {
    return {
      nearService: currentWallet ? new SputnikNearService(currentWallet) : null,
      availableWallets: availableWallets.map(availableWallet =>
        availableWallet.walletMeta()
      ),
      accountId: currentWallet?.getAccountId() ?? '',
      currentWallet: currentWallet?.getWalletType() ?? null,
      connectingToWallet,
      availableAccounts,
      pkAndSignature,
      login,
      logout,
      switchAccount,
      switchWallet,
    };
  }, [
    availableAccounts,
    availableWallets,
    connectingToWallet,
    currentWallet,
    login,
    logout,
    pkAndSignature,
    switchAccount,
    switchWallet,
  ]);

  return (
    <WalletContext.Provider value={walletContext}>
      {children}
      {connectingToWallet && (
        <ConnectingWalletModal
          isOpen
          onClose={() => toggleConnection(false)}
          walletType={walletContext?.currentWallet ?? WalletType.NEAR}
        />
      )}
    </WalletContext.Provider>
  );
};

export function useWalletContext(): WalletContext {
  return useContext(WalletContext);
}
