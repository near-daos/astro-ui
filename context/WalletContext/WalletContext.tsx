import {
  FC,
  useMemo,
  useState,
  useEffect,
  useContext,
  useCallback,
  createContext,
} from 'react';
import get from 'lodash/get';
import { useBoolean } from 'react-use';
import { useRouter } from 'next/router';

import { WalletType } from 'types/config';
import {
  WalletMeta,
  WalletService,
} from 'services/sputnik/SputnikNearService/walletServices/types';

import { configService } from 'services/ConfigService';
import { CookieService } from 'services/CookieService';
import { ACCOUNT_COOKIE } from 'constants/cookies';
import { SputnikNearService } from 'services/sputnik';
import { ConnectingWalletModal } from 'astro_2.0/features/Auth/components/ConnectingWalletModal';

import { GA_EVENTS, sendGAEvent } from 'utils/ga';
import { isSelectorWalletType } from 'utils/isSelectorWalletType';

import {
  NEAR_WALLET_METADATA,
  SENDER_WALLET_METADATA,
} from 'services/sputnik/SputnikNearService/walletServices/constants';

import { PkAndSignature } from './types';

import { useWallet } from './hooks/useWallet';
import { usePkAndSignature } from './hooks/usePkAndSignature';
import { useAvailableAccounts } from './hooks/useAvailableAccounts';
import { useSelectorLsAccount } from './hooks/walletSelector/useSelectorLsAccount';

export interface WalletContext {
  availableWallets: WalletMeta[];
  currentWallet: WalletType | null;
  accountId: string;
  connectingToWallet: boolean;
  login: (walletType: WalletType) => Promise<void>;
  logout: () => Promise<void>;
  switchAccount: (walletType: WalletType, accountId: string) => void;
  switchWallet: (walletType: WalletType) => void;
  availableAccounts: string[];
  pkAndSignature: PkAndSignature | null;
  // todo get rid of
  nearService: SputnikNearService | null;
}

const WalletContext = createContext<WalletContext>({} as WalletContext);

export const WrappedWalletContext: FC = ({ children }) => {
  const { nearConfig } = configService.get();

  const router = useRouter();

  const [, , removeSelectorLsAccount] = useSelectorLsAccount();
  const [connectingToWallet, setConnectingToWallet] = useBoolean(false);

  const { getWallet, setWallet, currentWallet, removePersistedWallet } =
    useWallet();

  const pkAndSignature = usePkAndSignature(currentWallet);
  const availableAccounts = useAvailableAccounts(currentWallet);
  const [currentAccount, setCurrentAccount] = useState('');

  useEffect(() => {
    async function updateAccountId() {
      if (currentWallet) {
        const account = await currentWallet.getAccountId();

        setCurrentAccount(account);
      }
    }

    updateAccountId();
  }, [currentWallet]);

  const signIn = useCallback(
    async (wallet: WalletService, contractName: string) => {
      setConnectingToWallet(true);

      try {
        await wallet.signIn(contractName);

        setWallet(wallet);

        setConnectingToWallet(false);

        const accountId = await wallet.getAccountId();

        sendGAEvent({
          name: GA_EVENTS.SIGN_IN,
          accountId,
        });
      } catch (e) {
        setConnectingToWallet(false);
      }
    },
    [setWallet, setConnectingToWallet]
  );

  const login = useCallback(
    async (walletType: WalletType) => {
      const wallet = await getWallet(walletType);

      if (!wallet) {
        return;
      }

      await signIn(wallet, nearConfig.contractName);
    },
    [signIn, getWallet, nearConfig.contractName]
  );

  const logout = useCallback(async () => {
    const accountId = (await currentWallet?.getAccountId()) ?? '';

    sendGAEvent({
      name: GA_EVENTS.SIGN_OUT,
      accountId,
    });

    CookieService.remove(ACCOUNT_COOKIE, { path: '/' });
    removePersistedWallet();

    await currentWallet?.logout();

    router.reload();
  }, [currentWallet, removePersistedWallet, router]);

  const switchAccount = useCallback(
    async (walletType: WalletType, accountId: string) => {
      // Currently its NEAR wallet specific only
      if (walletType !== WalletType.NEAR) {
        return;
      }

      const nearWallet = await getWallet(WalletType.NEAR);

      if (!nearWallet) {
        return;
      }

      // eslint-disable-next-line no-underscore-dangle
      const keypair = await nearWallet
        .getKeyStore()
        .getKey(nearConfig.networkId, accountId);

      if (!keypair) {
        return;
      }

      const authData = {
        accountId,
        allKeys: [keypair.getPublicKey().toString()],
      };

      // new wallet instance will take the new auth_key and will reinit the account
      window.localStorage.setItem(
        'sputnik_wallet_auth_key',
        JSON.stringify(authData)
      );

      CookieService.set(ACCOUNT_COOKIE, accountId, {
        path: '/',
      });

      sendGAEvent({
        name: GA_EVENTS.SWITCH_ACCOUNT,
        accountId,
      });

      router.reload();
    },
    [getWallet, nearConfig.networkId, router]
  );

  const switchWallet = useCallback(
    async (walletType: WalletType) => {
      const selectedWallet = await getWallet(walletType);

      if (!selectedWallet || !currentWallet) {
        return;
      }

      await signIn(selectedWallet, nearConfig.contractName);

      const currentWalletAccountId = await currentWallet.getAccountId();
      const selectedWalletAccountId = await selectedWallet.getAccountId();

      sendGAEvent({
        name: GA_EVENTS.SWITCH_WALLET,
        accountId: selectedWalletAccountId,
        params: {
          wallet: walletType,
          previousAccountId: currentWalletAccountId,
        },
      });

      if (currentWalletAccountId !== selectedWalletAccountId) {
        if (isSelectorWalletType(currentWallet.getWalletType())) {
          removeSelectorLsAccount();
        }

        router.reload();
      }
    },
    [
      router,
      signIn,
      getWallet,
      currentWallet,
      nearConfig.contractName,
      removeSelectorLsAccount,
    ]
  );

  const walletContext = useMemo(() => {
    const availableWallets = [NEAR_WALLET_METADATA];

    const senderWalletAvailable = get(window, 'near.isSender') || false;

    if (senderWalletAvailable) {
      availableWallets.push(SENDER_WALLET_METADATA);
    }

    return {
      login,
      logout,
      switchWallet,
      switchAccount,
      pkAndSignature,
      availableWallets,
      availableAccounts,
      connectingToWallet,
      accountId: currentAccount,
      currentWallet: currentWallet?.getWalletType() ?? null,
      nearService: currentWallet ? new SputnikNearService(currentWallet) : null,
    };
  }, [
    login,
    logout,
    switchWallet,
    switchAccount,
    currentWallet,
    pkAndSignature,
    currentAccount,
    availableAccounts,
    connectingToWallet,
  ]);

  return (
    <WalletContext.Provider value={walletContext}>
      {children}
      {connectingToWallet && (
        <ConnectingWalletModal
          isOpen
          onClose={() => setConnectingToWallet(false)}
          walletType={walletContext?.currentWallet ?? WalletType.NEAR}
        />
      )}
    </WalletContext.Provider>
  );
};

export function useWalletContext(): WalletContext {
  return useContext(WalletContext);
}
