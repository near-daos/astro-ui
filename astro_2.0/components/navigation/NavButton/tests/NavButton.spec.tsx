/* eslint-disable @typescript-eslint/ban-ts-comment */
import { render } from 'jest/testUtils';
import { useRouter } from 'next/router';
import { fireEvent } from '@testing-library/dom';

import { useWalletContext } from 'context/WalletContext';

import {
  NavButton,
  NavButtonProps,
} from 'astro_2.0/components/navigation/NavButton';

jest.mock('next/router', () => {
  return {
    ...jest.requireActual('next/router'),
    useRouter: jest.fn(() => ({
      query: {},
    })),
  };
});

jest.mock('context/WalletContext', () => {
  return {
    useWalletContext: jest.fn(() => ({})),
  };
});

jest.mock('components/Icon', () => {
  return {
    Icon: ({ name }: { name: string }) => <div>{name}</div>,
  };
});

describe('Nav button', () => {
  function renderNavButton(props?: Partial<NavButtonProps>) {
    return render(
      <NavButton
        myDaosIds={[]}
        icon="aAllDaos"
        hoverIcon="aAllDaosHover"
        href="somehref"
        label="Hello World"
        {...props}
      />
    );
  }

  it('Should render component', () => {
    const { container } = renderNavButton();

    expect(container).toMatchSnapshot();
  });

  it('Should navigate on click without authentication and when auth not required', () => {
    const href = 'helloWorld';

    const router = {
      query: {},
      push: jest.fn(),
    };

    // @ts-ignore
    useRouter.mockImplementation(() => router);

    const { getByRole } = renderNavButton({ href });

    fireEvent.click(getByRole('button'));

    expect(router.push).toBeCalledWith(href);
  });

  it('Should call login if navigation requires logged user and user is not logged', () => {
    const login = jest.fn();

    // @ts-ignore
    useWalletContext.mockImplementation(() => ({ login }));

    const { getByRole } = renderNavButton({ authRequired: true });

    fireEvent.click(getByRole('button'));

    expect(login).toBeCalled();
  });

  it('Should render "hover" icon on hover', () => {
    const { getByRole, getAllByText } = renderNavButton({ authRequired: true });

    fireEvent.mouseOver(getByRole('button'));

    expect(getAllByText('aAllDaosHover')).toHaveLength(1);
  });
});
