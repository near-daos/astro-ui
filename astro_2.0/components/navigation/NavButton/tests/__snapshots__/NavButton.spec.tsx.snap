// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav button Should render component 1`] = `
<div>
  <div
    class="root"
    role="button"
    tabindex="0"
  >
    <div
      class="description"
    >
      <div
        class="iconHolder"
      >
        <div>
          aAllDaos
        </div>
      </div>
      <div
        class="label"
      >
        Hello World
      </div>
    </div>
    <div
      class="underline"
    />
  </div>
</div>
`;
